### DISCUSSION TITLE
Project roadmap?

### DISCUSSION THREAD

**Discussion by sfre-coop** [microsoft/typespec#3092] [General]
I've started kicking the tyres on TypeSpec, not only because it looks a like a lot less typing than swagger <grin>, but also because a bit of type-checking can go a long way. It looks very promising.

I'm aware that this is relatively early days and not all initiatives survive. Is there a publicly-visible roadmap of where you intend to go with this? And, to be blunt, how much can I rely on it continuing to be around? Apologies if that sounds ungrateful.

Thanks for your work.

**Comment by bterlson** [comment]
Thanks for the words of encouragement!

I will point out that Azure has heavily adopted TypeSpec, and unwinding that investment would be very costly at this point. So while I can't predict the future, I think signals are that we're in it for the long haul.

**Reply by sfre-coop** [reply]
Thanks. good to hear. I suggest putting some words to that effect on the website. 

It's hard to tell from the outside how solid a project is and adopters need some sense of security before promoting it at our end.

**Reply by dstesaxobank** [reply]
@sfre-coop For what it's worth, the OpenAPI team at Saxo Bank is currently vetting TypeSpec as a replacement for our "handwritten" OpenAPI specifications and custom tooling. Our results look promising so far. If we decide to adopt TypeSpec, I expect members of my team — myself included — to actively contribute to the project.

@bterlson Thank you for your and your colleagues hard work.
